design/addon-categories.md

```diff
@@ -0,0 +1,132 @@
+# Add-On Compatibility and Conflicts
+
+This proposal defines a set of categories that add-ons can belong to and assigns each add-on to one or more categories.
+It also defines the dependencies of each add-on.
+
+## Goals
+
+- Offer high-level guidance to authors of kURL specs about which add-ons to include in their installer.
+- Decrease the likelihood of designing kURL specs that result in non-functioning installations.
+
+## Non Goals
+
+- Detecting syntax and low-level errors within the configurations of individual add-ons.
+
+## Background
+
+Authors of kURL specs must select from a growing list of add-ons.
+They may not be familiar with the function of the add-on and may not be aware of its dependencies.
+Authors must spend time researching each add-on and try installations to verify their spec results in a working cluster.
+
+A better experience would be to provide guidance that would lead to working clusters by default.
+
+## High-Level Design
+
+Each add-on belongs to one or more of the following categories:
+
+- "Container Runtime"
+- "CNI Plugin"
+- "PVC Provisioner"
+- "Object Store"
+- "Ingress"
+- "Logs"
+- "Snapshots"
+- "Metrics & Monitoring"
+- "Cluster Administration"
+- "Application Management"
+
+Each add-on can specify dependencies on other categories.
+
+Add-ons belonging to the same category conflict with each other.
+
+## Detailed Design
+
+Add a GET /add-ons endpoint that returns a list of add-ons along with their dependencies.
```

Comment:
@GraysonNull will you add some detail here describing how this endpoint will be implemented?  Is there a database or a file, or where does the data come from and how can we update it?

Reply:
Yeah....we need to figure that out. It seems pretty straight forward for us to generate a JSON file at build time and just serve it up all on the FE to avoid making more requests and the need for a new DB or database tables. My only concern with that is how do we add to it. Like is there a giant yaml file authors have to add to? Just feels brittle so I'm wondering if an API request to a database would be more stable over time as more add-ons are supported. 

I think a discussion with @areed and @mzaneri would be helpful in getting to a good solution for this.